File: GroceryListManager.py
class GroceryListManager:
    items = []

    def add_item(self):
        item = input("ENTER AN ITEM TO ADD: ")
        self.items.append(item)
        print(f"{item} added successfully")
        self.run()

    def remove_item(self):
        if len(self.items) == 0:
            print("NO ITEMS ARE AVAILABLE.")
            self.run()
            return
        print("LIST OF ALL AVAILABLE ITEMS:")
        for index, item in enumerate(self.items):
            print(f"{index + 1}. {item}")
        index = input("ENTER THE NUMBER OF THE ITEM TO REMOVE: ")
        if index.isdigit() and 0 < int(index) <= len(self.items):
            removed_item = self.items.pop(int(index) - 1)
            print(f"{removed_item} removed successfully")
        else:
            print("INVALID ITEM NUMBER!")
        self.run()

    def show_items(self):
        if len(self.items) == 0:
            print("NO ITEMS ARE AVAILABLE")
        else:
            print("Available items:")
            for index, item in enumerate(self.items):
                print(f"{index + 1}. {item}")
        self.run()

    def run(self):
        while True:
            prompt = """ WELCOME TO YOUR GROCERY LIST MANAGER
Press: 
1 -> Add item
2 -> Remove item
3 -> Show all items
0 -> Exit """
            print(prompt)
            choice = input("Kindly enter a choice: ")
            if choice == "1":
                self.add_item()
                break
            elif choice == "2":
                self.remove_item()
                break
            elif choice == "3":
                self.show_items()
                break
            elif choice == "0":
                print("NOW EXITING...")
                break
            else:
                print("INVALID CHOICE SELECTION!")

    def start(self):
        self.run()

manager = GroceryListManager()
manager.start()

File: DAY4/GroceryListManager.java
import java.util.ArrayList;
import java.util.Scanner;

public class GroceryListManager {
    private ArrayList<String> items;

    public GroceryListManager() {
        items = new ArrayList<>();
}

public void addItem() {
        Scanner scanner = new Scanner(System.in);
        System.out.print("ENTER AN ITEM TO ADD: ");
        String input = scanner.nextLine();
        items.add(input);
        System.out.println(input + " added successfully");
}

    public void removeItem() {
        if (items.isEmpty()) {
            System.out.println("NO ITEMS ARE AVAILABLE.");
            return;
}

        System.out.println("LIST OF All AVAILABLE ITEMS: ");
        for (int index = 0; index < items.size(); index++) {
            System.out.println((index + 1) + ". " + items.get(index));
}

        Scanner scanner = new Scanner(System.in);
        System.out.print("ENTER THE NUMBER OF THE ITEM TO REMOVE: ");
        int index = scanner.nextInt();
	scanner.nextLine();

        if (index > 0 && index <= items.size()) {
            String removedItem = items.remove(index - 1);
            System.out.println(removedItem + " removed successfully");
} else {
            System.out.println("INVALID ITEM NUMBER!");
        }
    }

    public void showItems() {
        if (items.isEmpty()) {
            System.out.println("NO ITEMS ARE AVAILABLE");
        } else {
            System.out.println("Available items:");
            for (int index = 0; index < items.size(); index++) {
                System.out.println((index + 1) + ". " + items.get(index));
}
}
}

    public void run() {

        while (true) {
            String prompt = """
                    WELCOME TO YOUR GROCERY LIST MANAGER

                    Press:
                        1 -> Add item
                        2 -> Remove item
                        3 -> Show all items
                        0 -> Exit
                    """;

            System.out.println(prompt);
            System.out.print("Kindly enter a choice: ");
            Scanner scanner = new Scanner(System.in);
            int choice = scanner.nextInt();
		scanner.nextLine();
            
            switch (choice) {
                case 1:
                    addItem();
                    break;
                case 2:
                    removeItem();
                    break;
                case 3:
                    showItems();
                    break;
                case 0:
                    System.out.println("NOW EXITING...");
                    return;
                default:
                    System.out.println("INVALID CHOICE SELECTION!");
 }
 }
 }

    public static void main(String[] args) {
        GroceryListManager manager = new GroceryListManager();
        manager.run();
}
}